### PR TITLE
Test cases for new ReferenceError: remove class tests and add no-strict

### PR DESCRIPTION
**Pull request by spectranaut**
Follow up from: https://github.com/tc39/test262/pull/2447#issuecomment-571207108 

This PR:
- deletes the auto-generated class tests that don't apply
- adds a new set of auto-generation templates for tests that only error in sloppy mode, which includes all the function forms (with the flag "no-strict) except for classes (see `src/function-forms/error-no-strict`)
- generates the reference error tests for the no-strict templates only

Thanks @patrick-soquet for noticing the incorrect class tests!  Want to take a look, @syg?

### CODE REVIEW

File: test/language/expressions/async-arrow-function/eval-var-scope-syntax-err.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
 // This file was procedurally generated from the following sources:
 // - src/function-forms/eval-var-scope-syntax-err.case
-// - src/function-forms/error/async-arrow-function.template
+// - src/function-forms/error-no-strict/async-arrow-function.template
 /*---
-description: sloppy direct eval in params introduces var (async arrow function expression)
+description: sloppy direct eval in params introduces var (async arrow function expression in sloppy code)
 esid: sec-async-arrow-function-definitions
 features: [default-parameters, async-functions]
-flags: [generated, noStrict, async]
```
Comment (syg) [inline test/language/expressions/async-arrow-function/eval-var-scope-syntax-err.js]: The `noStrict` flag is incorrectly removed here due to flags misformatting in the template.

File: test/language/expressions/async-function/named-eval-var-scope-syntax-err.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
 // This file was procedurally generated from the following sources:
 // - src/function-forms/eval-var-scope-syntax-err.case
-// - src/function-forms/error/async-func-expr-named.template
+// - src/function-forms/error-no-strict/async-func-expr-named.template
 /*---
-description: sloppy direct eval in params introduces var (async function named expression)
+description: sloppy direct eval in params introduces var (async function named expression in sloppy code)
 esid: sec-async-function-definitions
 features: [default-parameters, async-functions]
-flags: [generated, noStrict, async]
```
Comment (syg) [inline test/language/expressions/async-function/named-eval-var-scope-syntax-err.js]: Ditto.

File: test/language/expressions/async-function/nameless-eval-var-scope-syntax-err.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
 // This file was procedurally generated from the following sources:
 // - src/function-forms/eval-var-scope-syntax-err.case
-// - src/function-forms/error/async-func-expr-nameless.template
+// - src/function-forms/error-no-strict/async-func-expr-nameless.template
 /*---
-description: sloppy direct eval in params introduces var (async function nameless expression)
+description: sloppy direct eval in params introduces var (async function nameless expression in sloppy code)
 esid: sec-async-function-definitions
 features: [default-parameters, async-functions]
-flags: [generated, noStrict, async]
```
Comment (syg) [inline test/language/expressions/async-function/nameless-eval-var-scope-syntax-err.js]: Ditto.

File: test/language/expressions/object/method-definition/async-meth-eval-var-scope-syntax-err.js [hunk 4]
```diff
@@ -1,11 +1,11 @@
 // This file was procedurally generated from the following sources:
 // - src/function-forms/eval-var-scope-syntax-err.case
-// - src/function-forms/error/async-meth.template
+// - src/function-forms/error-no-strict/async-meth.template
 /*---
-description: sloppy direct eval in params introduces var (async method)
+description: sloppy direct eval in params introduces var (async method in sloppy code)
 esid: sec-async-function-definitions
 features: [default-parameters, async-functions]
-flags: [generated, noStrict, async]
```
Comment (syg) [inline test/language/expressions/object/method-definition/async-meth-eval-var-scope-syntax-err.js]: Ditto.

File: test/language/statements/async-function/eval-var-scope-syntax-err.js [hunk 5]
```diff
@@ -1,11 +1,11 @@
 // This file was procedurally generated from the following sources:
 // - src/function-forms/eval-var-scope-syntax-err.case
-// - src/function-forms/error/async-func-decl.template
+// - src/function-forms/error-no-strict/async-func-decl.template
 /*---
-description: sloppy direct eval in params introduces var (async function declaration)
+description: sloppy direct eval in params introduces var (async function declaration in sloppy code)
 esid: sec-async-function-definitions
 features: [default-parameters, async-functions]
-flags: [generated, noStrict, async]
```
Comment (syg) [inline test/language/statements/async-function/eval-var-scope-syntax-err.js]: Ditto.

File: src/function-forms/error-no-strict/async-arrow-function.template [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+  Supplemental Syntax
+
+  When processing an instance of the production AsyncArrowFunction :
+  CoverCallExpressionAndAsyncArrowHead => AsyncConciseBody the interpretation of
+  CoverCallExpressionAndAsyncArrowHead is refined using the following grammar:
+
+  AsyncArrowHead :
+    async ArrowFormalParameters
+flags: [async]
```
Comment (syg) [inline src/function-forms/error-no-strict/async-arrow-function.template]: The two `flag:` sections should be combined. The first `noStrict` flags section is being ignored.

File: src/function-forms/error-no-strict/async-func-expr-nameless.template [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+path: language/expressions/async-function/nameless-
+name: async function nameless expression in sloppy code
+esid: sec-async-function-definitions
+flags: [noStrict]
+info: |
+  14.6 Async Function Definitions
+
+  AsyncFunctionExpression :
+    async function ( FormalParameters ) { AsyncFunctionBody }
+flags: [async]
```
Comment (syg) [inline src/function-forms/error-no-strict/async-func-expr-nameless.template]: Ditto.

File: src/function-forms/error-no-strict/async-func-expr-named.template [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+path: language/expressions/async-function/named-
+name: async function named expression in sloppy code
+esid: sec-async-function-definitions
+flags: [noStrict]
+info: |
+  14.6 Async Function Definitions
+
+  AsyncFunctionExpression :
+    async function BindingIdentifier ( FormalParameters ) { AsyncFunctionBody }
+flags: [async]
```
Comment (syg) [inline src/function-forms/error-no-strict/async-func-expr-named.template]: Ditto.

File: src/function-forms/error-no-strict/async-func-decl.template [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+path: language/statements/async-function/
+name: async function declaration in sloppy code
+esid: sec-async-function-definitions
+flags: [noStrict]
+info: |
+  14.6 Async Function Definitions
+
+  AsyncFunctionDeclaration :
+    async function BindingIdentifier ( FormalParameters ) { AsyncFunctionBody }
+flags: [async]
```
Comment (syg) [inline src/function-forms/error-no-strict/async-func-decl.template]: Ditto.

File: src/function-forms/error-no-strict/async-meth.template [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+path: language/expressions/object/method-definition/async-meth-
+name: async method in sloppy code
+esid: sec-async-function-definitions
+flags: [noStrict]
+info: |
+  14.6 Async Function Definitions
+
+  AsyncMethod :
+   async PropertyName ( UniqueFormalParameters ) { AsyncFunctionBody }
+flags: [async]
```
Comment (syg) [inline src/function-forms/error-no-strict/async-meth.template]: Ditto.

**Comment by spectranaut** [comment]
thanks for catching that, @syg, fixed!

**Pull request merged**